### PR TITLE
Fix yaml serialization in io mixin

### PR DESCRIPTION
**Pull request by hemildesai**
Uses __class__ only when data.__qualname__ or similar fails

### CODE REVIEW

File: nemo/lightning/io/mixin.py [hunk 1]
```diff
@@ -81,15 +81,17 @@ def _partial_representer_with_defaults(dumper, data):
...
 def _safe_object_representer(dumper, data):
-    if not inspect.isclass(data):
-        cls = data.__class__
-        call = True
-    else:
+    try:
         cls = data
+        target = f"{inspect.getmodule(cls).__name__}.{cls.__qualname__}"
         call = False
+    except Exception:
```
Comment (cuichenx) [inline nemo/lightning/io/mixin.py:88]: is it possible to specify the exception types here? or is it difficult to enumerate

**Review by cuichenx — COMMENTED** [review]
Would be good if you could add some comments or doc strings to this function too, thanks!

File: nemo/lightning/io/mixin.py (hunk 1 above)
Comment (hemildesai) [reply nemo/lightning/io/mixin.py:88]: I can change it to `AttributeError`

**Comment by hemildesai** [comment]
> Add docstring

Updated in https://github.com/NVIDIA/NeMo/pull/11106/commits/bd903a0c86b8ba847903f9ba8e1aad37f1f5ea6a

File: nemo/lightning/io/mixin.py (hunk 1 above)
Comment (hemildesai) [reply nemo/lightning/io/mixin.py]: Changed in https://github.com/NVIDIA/NeMo/pull/11106/commits/bd903a0c86b8ba847903f9ba8e1aad37f1f5ea6a

**Review by oyilmaz-nvidia — APPROVED** [review]

**Comment by pablo-garay** [comment]
CI passed : https://github.com/NVIDIA/NeMo/actions/runs/11602055172

**Comment by pablo-garay** [comment]
There was a further commit after the triggered CI run but it was a docstring name + name change which shouldnt affect result:
https://github.com/NVIDIA/NeMo/pull/11106/commits/bd903a0c86b8ba847903f9ba8e1aad37f1f5ea6a

**Pull request merged**